java/code/src/com/redhat/rhn/frontend/xmlrpc/schedule/ScheduleHandler.java

```diff
@@ -219,6 +219,25 @@ public int failSystemAction(User loggedInUser, Integer serverId, Integer actionI
         return dr.toArray();
     }
 
+    /**
+     * List all the scheduled actions that have been archived.
+     * @param loggedInUser The current user
+     * @return Returns a list of actions with details
+     *
+     * @xmlrpc.doc Returns a list of actions that have been archived.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.returntype
+     * #array_begin()
+     *   $ScheduleActionSerializer
+     * #array_end()
+     */
+    public Object[] listAllArchivedActions(User loggedInUser) {
+        // the second argument is "PageControl". This is not needed for the api usage;
+        // therefore, null will be used.
+        DataResult dr = ActionManager.allArchivedActions(loggedInUser, null);
```

Comment:
```suggestion
        return ActionManager.allArchivedActions(loggedInUser, null).toArray();
```
Maybe do it this way and just drop next line?